src/Component.js

```diff
@@ -1,73 +1,51 @@
 /* Represents a component of a vector */
-class Component
-{
-	constructor(length, maxRep, jump, hop, promote)
-	{
+class Component {
```

Comment:
nit: I drop my open braces. We should decide if we care about keeping that consistent across the repo, always leave them same line, or always drop them.